src/item.h

```diff
@@ -704,8 +704,9 @@ class item : public JsonSerializer, public JsonDeserializer
         /**
          * Callback when a player starts wielding the item. The item is already in the weapon
          * slot and is called from there.
+         * @param mv number of moves *already* spent wielding the weapon
          */
-        void on_wield( player &p );
+        void on_wield( player &p, int mv = 0 );
```

Comment:
The message-generating code would need to be duplicated in `player::wield_contents`. That said that could be also be an opportunity to provide more descriptive messages *"You draw your foo from your quux"*